OpenRA.Mods.Common/Traits/Player/ProductionQueue.cs

```diff
@@ -102,14 +105,20 @@ public ProductionQueue(ActorInitializer init, Actor playerActor, ProductionQueue
 			developerMode = playerActor.Trait<DeveloperMode>();
 
 			Faction = init.Contains<FactionInit>() ? init.Get<FactionInit, string>() : self.Owner.Faction.InternalName;
-			Enabled = !info.Factions.Any() || info.Factions.Contains(Faction);
+			ValidFaction = !info.Factions.Any() || info.Factions.Contains(Faction);
+			Enabled = ValidFaction;
```

Comment:
In a future PR we could consider replacing the special case faction logic with a faction-granted condition.